src/transformers/hf_argparser.py

```diff
@@ -43,6 +54,60 @@ def string_to_bool(v):
         )
 
 
+def make_choice_type_function(choices):
+    str_to_choice = {str(choice): choice for choice in choices}
+    return lambda arg: str_to_choice.get(arg, arg)
+
+
+def HfArg(
+    *,
+    aliases: Union[str, List[str]] = None,
+    help: str = None,
+    default: Any = dataclasses.MISSING,
+    default_factory: Callable[[], Any] = dataclasses.MISSING,
+    metadata: dict = None,
+    **kwargs,
+):
+    """
+    Argument helper enabling a concise syntax to create dataclass fields for parsing with `HfArgumentParser`.
+
+    Example comparing the use of `HfArg` and `dataclasses.field`:
+    ```
+    @dataclass
+    class Args:
+        regular_arg: str = dataclasses.field(default="Huggingface", metadata={"aliases": ["--example", "-e"], "help": "This syntax could be better!"})
+        hf_arg: str = HfArg(default="Huggingface", aliases=["--example", "-e"], help="What a nice syntax!")
+    ```
+
+    Args:
+        aliases:
+            Single string or list of strings of aliases to pass on to argparse, e.g. `aliases=["--example", "-e"]`
+        help:
+            Help string to pass on to argparse that can be displayed with --help.
+        default:
+            Default value. If not default or default_factory is specified, the argument is required.
+        default_factory:
+            The default_factory is a 0-argument function called to initialize a field's value. It is useful to provide
+            default values for mutable types, e.g. lists: `default_factory=list`
+        metadata:
+            Further metadata to pass on to `dataclasses.field`.
+        kwargs:
+            (Optional) Passed on to `dataclasses.field`.
```

Comment:
Would be great to document the defaults/types here using our regular format:
```
<arg_name> (<type>, *optional*, defaults to <default>):
```
I know the rest of the file does not follow it but it's old doc :-)